enhancements/network/optional-vips-with-extlb.md

```diff
@@ -0,0 +1,204 @@
+---
+
+# Optional VIPs with External Loadbalancer
+
+## Summary
+
+Stop requiring VIP configuration when an external loadbalancer is in use.
+This will allow external loadbalancer customers to use their own DNS as well,
+effectively disabling most of the internal DNS and LB infrastructure.
+
+## Motivation
+
+Previously when an external loadbalancer was configured we still used the VIP
+fields of install-config to create DNS records in the internal DNS
+infrastructure pointing at the external loadbalancer. However, some more
+advanced loadbalancers also use DNS loadbalancing to further distribute
+traffic. This does not work if we hard-code a single loadbalancer endpoint
+into our internal DNS.
+
+### User Stories
+
+* As an OpenShift deployer I want to use an advanced external loadbalancer
+  to distribute traffic to my cluster. This loadbalancer uses multiple DNS
+  targets in order to reduce load on a single endpoint.
+
+### Goals
+
+Make it possible to disable the internal DNS records for api, api-int, and
+ingress in on-prem IPI deployments to allow use of external DNS to manage
+those, even for internal cluster traffic.
+
+### Non-Goals
+
+Disabling the internal DNS infrastructure entirely. There are two reasons
+we're not doing this:
+
+* It also provides records for each host in the cluster, although it's not
+  clear this is actually necessary anymore.
+* It would require significant re-working of the resolv-prepender behavior,
+  and that component has historically been rather fragile. Adding complexity
+  in that area is more likely to cause problems and would run counter to past
+  efforts to simplify it.
+
+## Proposal
+
+When an external loadbalancer is configured we will stop requiring VIPs to be
+specified for creating DNS records. If VIPs are not provided, we will not
+configure any DNS records for the api, api-int, and ingress addresses. The
+deployer will then be wholly responsible for providing those records.
+
+Note that api and ingress are already required external records, so in reality
+the only addition here is api-int.
+
+### Workflow Description
+
+The deployer will configure external DNS and loadbalancing, similar to how
+they would for a UPI deployment. They will then deploy an IPI cluster with
+the loadbalancer set to UserManaged and no VIPs configured.
+
+### API Extensions
+
+The VIP fields will no longer be mandatory for all on-prem IPI deployments.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No Hypershift impacts
+
+#### Standalone Clusters
+
+Yes, this is expected to be used primarily with standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+SNO is not supported for on-prem IPI deployments, so no impacts there.
+
+I don't believe this will impact MicroShift either, but even if it did it would
+reduce resources requirements slightly.
+
+### Implementation Details/Notes/Constraints
+
+A configuration to use this new functionality would look like the following:
```

Comment:
What configuration is this? Who is creating it? What is reading it?

Reply:
This snippet comes from the install-config.yaml